src/styles/main.scss

```diff
@@ -1,3 +1,60 @@
+@import './variables';
+
 body {
   margin: 0;
+  box-sizing: border-box;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  color: black;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  width: $calendar-width;
+  padding: $calendar-padding;
+  margin: 0 auto;
+
+  @for $length from 28 through 31 {
+    &--month-length-#{$length} {
+      .calendar__day:nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $day, $idx in $days-of-week {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: $idx * ($cell-width + $calendar-gap + 2);
+      }
+    }
+  }
+
+  &__day {
+    border: 1px solid black;
+    background-color: #eee;
+    height: $cell-height;
+    width: $cell-width;
+    text-align: center;
+    line-height: 100px;
+
+    @for $days from 1 through 31 {
+      &:nth-child(#{$days})::before {
+        content: '#{$days}';
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transition: all 0.5s;
```

Comment:
It's better to place `transition` property inside the original element selector (`&__day`) but not inside the `:hover` state. So you will have a smooth transition on hover as well as on unhover.